Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -297,10 +297,9 @@ open class NavigationMapView: UIView {
     }
     
     func setupGestureRecognizers() {
-        let gestures = gestureRecognizers ?? []
-        let mapTapGesture = UITapGestureRecognizer(target: self, action: #selector(didRecieveTap(sender:)))
-        mapTapGesture.requireFailure(of: gestures)
-        mapView.addGestureRecognizer(mapTapGesture)
+        for recognizer in mapView.gestureRecognizers ?? [] where recognizer is UITapGestureRecognizer {
```

Comment:
Do we need to add a target to all of the `UITapGestureRecognizer` instances?

Reply:
We technically don't, but the gesture recognizers from the Maps SDK aren't named, so there's no way to filter through them effectively. The only way would be through target actions but those are private instance variables. That approach would either be fragile or rejected by Apple's review process. Trying to add to a certain gesture recognizer by order (ex// trying to add target to the last gesture recognizer in the `mapView.gestureRecognizer` array) also isn't very reliable. This fix is potentially temporary for now anyway until the Maps team implements logic on their side.

Reply:
@jill-cardamon, can you please create a follow-up ticket for this? If this is temporary workaround we should revise current implementation. If I understand this correctly idea would be to register only one gesture recognizer on Navigation SDK side, which will allow to select routes and waypoints. Such gesture recognizer is supposed to fail when such mechanism is available on Maps SDK side, right? 

Please also squash current PR into one commit, we do not want duplication in history based on previous PRs.

Reply:
It'd be also good to make sure that route selection logic works on Navigation SDK `1.x` and when custom gestures are added on client side.